NeoML/test/src/DnnBlobTest.cpp

```diff
@@ -16,28 +16,172 @@ limitations under the License.
 #include <common.h>
 #pragma hdrstop
 
+#include <atomic>
+#include <thread>
 #include <TestFixture.h>
 
-TEST(CDnnBlobTest, InitWindowBlob)
+using namespace NeoML;
+using namespace NeoMLTest;
+
+TEST( CDnnBlobTest, InitWindowBlob )
 {
-    NeoML::IMathEngine& engine = NeoML::GetDefaultCpuMathEngine();
-    CPtr<NeoML::CDnnBlob> parent = NeoML::CDnnBlob::CreateDataBlob( engine, NeoML::CT_Float, 16, 1, 1 );
-    CPtr<NeoML::CDnnBlob> blob = NeoML::CDnnBlob::CreateWindowBlob( parent );
+    CPtr<CDnnBlob> parent = CDnnBlob::CreateDataBlob( MathEngine(), CT_Float, 16, 1, 1 );
+    CPtr<CDnnBlob> blob = CDnnBlob::CreateWindowBlob( parent );
 
     ASSERT_FALSE( blob->GetData().IsNull() );
 }
 
-TEST(CDnnBlobTest, BufferTest)
+TEST( CDnnBlobTest, BufferTest )
 {
-    NeoML::IMathEngine& engine = NeoML::GetDefaultCpuMathEngine();
-    CPtr<NeoML::CDnnBlob> blob = NeoML::CDnnBlob::CreateDataBlob( engine, NeoML::CT_Float, 16, 1, 1 );
+    CPtr<CDnnBlob> blob = CDnnBlob::CreateDataBlob( MathEngine(), CT_Float, 16, 1, 1 );
     ASSERT_FALSE( blob->GetData().IsNull() );
 
-    NeoML::CDnnBlobBuffer<float> buffer( *blob, NeoML::TDnnBlobBufferAccess::Write );
+    CDnnBlobBuffer<float> buffer( *blob, TDnnBlobBufferAccess::Write );
     ASSERT_NE( nullptr, buffer.Ptr() );
-    ::memset( buffer, 0, buffer.Size() * sizeof(float) );
+    ::memset( buffer, 0, buffer.Size() * sizeof( float ) );
 
     EXPECT_FALSE( buffer.IsClosed() );
     buffer.Close();
     EXPECT_TRUE( buffer.IsClosed() );
 }
+
+//---------------------------------------------------------------------------------------------------------------------
+
+#if FINE_PLATFORM( FINE_WINDOWS ) || !defined( NEOML_USE_FINEOBJ )
+
+namespace NeoMLTest {
+
+enum class TTransferType {
+    PoolToPool, PoolToHeap, HeapToPool, HeapToHeap
+};
+
+class CDnnBlobTest : public CNeoMLTestFixture, public ::testing::WithParamInterface<TTransferType> {
+public:
+	static bool InitTestFixture() { return true; }
+	static void DeinitTestFixture() {}
+};
+
+static void testTransferBlobInThreadsImpl( TTransferType type )
+{
+    const TMathEngineType met = MathEngine().GetType();
+    switch( met ) {
+        case MET_Cpu:
+            break;
+        case MET_Cuda:
+            if( TTransferType::PoolToPool == type ) {
+                break;
+            }
+        case MET_Metal:
```

Comment:
Modern compilers generate warnings when there's no break nor comment before switch case (they suspect us in forgetting `break`s)

Reply:
fixed